remotes/handlers.go

```diff
@@ -33,6 +33,37 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// ManifestHandler is used to handle image fetch or converter.
+type ManifestHandler interface {
+	Handle(ctx context.Context, desc ocispec.Descriptor) (ocispec.Descriptor, error)
+}
+
+// ManifestHandlerFunc is the implementation the ManifestHandler interface.
+type ManifestHandlerFunc func(ctx context.Context, desc ocispec.Descriptor) (ocispec.Descriptor, error)
+
+// Handle image fetch or converter.
+func (mh ManifestHandlerFunc) Handle(ctx context.Context, desc ocispec.Descriptor) (ocispec.Descriptor, error) {
+	return mh(ctx, desc)
+}
+
+// ManifestHandlers returns a handler which will run the handlers in order.
+func ManifestHandlers(handlers ...ManifestHandler) ManifestHandlerFunc {
+	return func(ctx context.Context, desc ocispec.Descriptor) (ocispec.Descriptor, error) {
+		var (
+			ndesc ocispec.Descriptor = desc
+			err   error
+		)
+
+		for _, h := range handlers {
+			ndesc, err = h.Handle(ctx, desc)
```

Comment:
Based on how this function is implemented, I would expect `ndesc` to be used by the next handler. Or in this case, just always overwrite `desc`.

Reply:
my mistake. will update it.